gui/melt-remainder.lua

```diff
@@ -0,0 +1,119 @@
+-- Add a display for how much melted metal is stored inside smelters.
+--@module = true
+--@enable = true
+--[====[
+
+gui/melt-remainder
+==================
+When enabled, a line of text is added to the Tasks screen of melters and magma smelters
+that shows how much metal is "stored" as a result of melting items in it. (The base game
+has no display of this information,)
+
+Click on the text for more detailed information.
+
+]====]
+
+local overlay = require("plugins.overlay")
+local widgets = require("gui.widgets")
+-- the existence of this is not documented anywhere of course
+local dialogs = require("gui.dialogs")
+
+ENABLED = ENABLED or false
+
+local function i_hate_lua(tbl)
+  local worst_language = 0
+  for _,_ in pairs(tbl) do
+    worst_language = worst_language + 1
+  end
+  return worst_language
+end
+
+local function get_melt_remainders(smelter)
+  if not smelter.melt_remainder then return nil end
+  local fractions = {}
+  local mat_count = #df.global.world.raws.inorganics
+  for i = 0, mat_count - 1 do
+    local melt_frac = smelter.melt_remainder[i]
+    if melt_frac > 0 then
+      fractions[i] = melt_frac
+    end
+  end
+  return fractions
+end
+
+-- lua doesn't hoist functions nerd emoji
+local function popup_full_list()
+  local workshop = dfhack.gui.getSelectedBuilding(true)
+  if not workshop then return end
+  local rems = get_melt_remainders(workshop)
+  if not rems then return end
+
+  printall(rems)
+  local lines = {}
+  for mat_id, tenths in pairs(rems) do
+    local mat_name = df.global.world.raws.inorganics[mat_id].id
```

Comment:
you should probably be using the material lookup functions for this: https://docs.dfhack.org/en/stable/docs/dev/Lua%20API.html#material-info-lookup

e.g. `dfhack.matinfo.decode(0, mat_id)`

then you can get the in-game name from the returned object.